### PR TITLE
ODIM 5644: The ACI plugin config password is currently plain text and should be encrypted

### PR DESCRIPTION
**Pull request by lpooja**
This PR contains changes for ACI plugin config password in encrypted format.

"OdimPassword" paramter in Aci config file should be passed in encrypted form now onwards.
Logic is handled on plugin side to decrypt it and validate the same.

### CODE REVIEW

File: caputilities/enigma.go [hunk 1]
```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2020 Intel Corporation
```
Comment (jeevan-kamkar) [inline caputilities/enigma.go]: change the header to HPE and update licence to 2022

File: caputilities/enigma.go [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+	"crypto/rsa"
+	"crypto/sha512"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+
+	logging "github.com/sirupsen/logrus"
+	//"github.com/ODIM-Project/ODIM/plugin-unmanaged-racks/logging"
```
Comment (jeevan-kamkar) [inline caputilities/enigma.go]: remove the commented code

File: caputilities/enigma.go [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+	return base64.StdEncoding.EncodeToString(encrypted)
+}
+
+func NewEnigma(privKeyPath string) (*Enigma, error) {
+	privateKeyBytes, err := ioutil.ReadFile(privKeyPath)
+	if err != nil {
+		return nil, fmt.Errorf("Cannot load PrivateKey from given path: '%s' because of  %s", privKeyPath, err)
+	}
+
+	// publicKeyBytes, err := ioutil.ReadFile(pubKeyPath)
```
Comment (jeevan-kamkar) [inline caputilities/enigma.go]: remove the commented code

File: caputilities/enigma.go [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+	// if err != nil {
+	// 	return nil, fmt.Errorf("Cannot load PublicKey from given path: '%s' because of %s", pubKeyPath, err)
+	// }
+
+	return &Enigma{
+		priv: bytesToPrivateKey(privateKeyBytes),
+		// pub:  bytesToPublicKey(publicKeyBytes),
```
Comment (jeevan-kamkar) [inline caputilities/enigma.go]: remove this if it's not used

File: caputilities/enigma_test.go [hunk 5]
```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2020 Intel Corporation
+ *
```
Comment (jeevan-kamkar) [inline caputilities/enigma_test.go:4]: change the header to hpe and update the year refer the odim code for the same

File: config/config.go [hunk 6]
```diff
@@ -90,10 +90,13 @@ type MessageBusConf struct {
 type KeyCertConf struct {
 	RootCACertificatePath string `json:"RootCACertificatePath"` // RootCACertificate will be added to truststore
 	PrivateKeyPath        string `json:"PrivateKeyPath"`        // plugin private key
+	PublicKeyPath         string `json:"PublicKeyPath"`         // plugin private key
```
Comment (jeevan-kamkar) [inline config/config.go]: rename it to the public key in the comment and whether this variable is used?

File: install/Kubernetes/helmcharts/aciplugin/templates/configmaps.yaml [hunk 7]
```diff
@@ -22,7 +22,8 @@ data:
     	"KeyCertConf": {
     		"RootCACertificatePath": "/etc/odimra_certs/rootCA.crt",
     		"PrivateKeyPath": "/etc/odimra_certs/odimra_server.key",
-    		"CertificatePath": "/etc/odimra_certs/odimra_server.crt"
+    		"CertificatePath": "/etc/odimra_certs/odimra_server.crt",
+				"RSAPrivateKeyPath": "/etc/odimra_certs/odimra_rsa.private"
```
Comment (jeevan-kamkar) [inline install/Kubernetes/helmcharts/aciplugin/templates/configmaps.yaml]: check the indentation

**Review by jeevan-kamkar — CHANGES_REQUESTED** [review]
address the  requested changes

File: caputilities/enigma.go (hunk 1 above)
Comment (lpooja) [reply caputilities/enigma.go]: changed

File: caputilities/enigma.go (hunk 2 above)
Comment (lpooja) [reply caputilities/enigma.go]: removed the comments

File: caputilities/enigma.go (hunk 3 above)
Comment (lpooja) [reply caputilities/enigma.go]: removed

File: caputilities/enigma.go (hunk 4 above)
Comment (lpooja) [reply caputilities/enigma.go]: removed

File: caputilities/enigma_test.go (hunk 5 above)
Comment (lpooja) [reply caputilities/enigma_test.go:4]: added

File: config/config.go (hunk 6 above)
Comment (lpooja) [reply config/config.go]: removed as it is not used for now

File: install/Kubernetes/helmcharts/aciplugin/templates/configmaps.yaml (hunk 7 above)
Comment (lpooja) [reply install/Kubernetes/helmcharts/aciplugin/templates/configmaps.yaml]: done

**Review by jeevan-kamkar — APPROVED** [review]

**Review by shivacharanms — APPROVED** [review]

**Review by hemkapoor — APPROVED** [review]

**Pull request merged**